### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by chunderstruck**
Added logging functionality, by default logs to /var/log/ccfvm.log
May need additional verification to make sure that we are getting proper console logging.  Not sure if that is going to matter in the long run but may be an issue while in development.

Can monitor logs while testing:
`tail -f /var/log/ccfvm.log`

Added to automation.sh
logging configuration file, this can be used in any other module that may need logging support. (GRPC?)

### CODE REVIEW

File: scripts/automation.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 
 echo "Getting Python dependencies"
-pip install boto3
+pip install -r requirements.txt
```
Comment (rough007) [inline scripts/automation.sh]: This should be additive, not replacing boto3

**Comment by rough007** [comment]
This is great!  Can you make so that it prints and logs each thing?  What's the easiest way to do that? Maybe an output function that takes text to both?  Maybe something like that exists already... idk but this looks good

**Comment by rough007** [comment]
Let me clarify. I want some sort of terminal feedback that the automation tool shows the user.

**Comment by chunderstruck** [comment]
`echo "Getting Python dependencies"
-pip install boto3
+pip install -r requirements.txt`

This should install all dependencies not just boto3.  Moved boto3 into requirements.txt so if desired later we can move the requirements into the build script.

**Comment by rough007** [comment]
Yeah, that makes sense

**Comment by chunderstruck** [comment]
Working on using a different configuration file format that may allow for more functionality... stand by...

**Comment by chunderstruck** [comment]
- Modified Loggers to handle all output within the DEBUG level to both console and LogFile
- Three Logging Levels are being used.
... Debug - Everything that would need to log console output
... INFO - All informative items, providing specifics on actions that are occuring.  (ie.  Specific file names, databases, usernames that are referenced)
... WARNING - Anything resulting in a warning message.


Currently only logging items in rc.py, logging levels can be modified in the configuration file so not logging every single action to the logfile or console.  By setting default logging levels to INFO, where as the current level for everything is DEBUG.

**Comment by chunderstruck** [comment]
Example of the output from log files.
screen shot 2018-03-24 at 2 22 57 pm

**Comment by rough007** [comment]
This looks good.  Merging

**Pull request merged**